src/sentry/notifications/platform/target.py

```diff
@@ -31,11 +41,29 @@ class NotificationTarget:
 
 
 @dataclass(kw_only=True, frozen=True)
-class IntegrationNotificationTarget(NotificationTarget):
+class IntegrationNotificationTarget(GenericNotificationTarget):
     """
     Adds necessary properties and methods to designate a target within an integration.
     Accepts the renderable object type that matches the connected provider.
     """
 
     integration_id: int
     organization_id: int
+    integration: RpcIntegration = field(init=False)
```

Comment:
q im kinda confused on the flow of this target class & `prepare_targets`? Is the scope of a single `IntegrationNotificationTarget` a everything under a single `OrganizationIntegration` ? I guess I'm confused on how these properties on the target work if there's potentially multiple `OrganizationIntegrations` attached to an `Integration`